### PR TITLE
[Feature] huasdorff distance loss

### PR DESCRIPTION
**Pull request by jinxianwei**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add Huasdorff distance loss

### CODE REVIEW

File: mmseg/models/losses/huasdorff_distance_loss.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            item to be included into the backward graph, `loss_` must be the
+            prefix of the name. Defaults to 'loss_boundary'.
+    """
+
+    def __init__(self,
+                 reduction='mean',
+                 class_weight=None,
+                 loss_weight=1.0,
+                 ignore_index=255,
+                 loss_name='loss_huasdorff_disstance',
```
Comment (xiexinch) [inline mmseg/models/losses/huasdorff_distance_loss.py:95]: Need to add type hints.

File: mmseg/models/losses/huasdorff_distance_loss.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            gt_dtm,
+            class_weight=None,
+            ignore_index=255):
+    """
+    compute huasdorff distance loss for segmentation
+    Args:
+        seg_soft: softmax results, shape=(b,c,x,y)
+        gt: ground truth, shape=(b,x,y)
+        seg_dtm: segmentation distance transform map, shape=(b,c,x,y)
+        gt_dtm: ground truth distance transform map, shape=(b,c,x,y)
```
Comment (xiexinch) [inline mmseg/models/losses/huasdorff_distance_loss.py:52]: Need to add type hints.

File: mmseg/models/losses/huasdorff_distance_loss.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+from mmseg.registry import MODELS
+from .utils import get_class_weight, weighted_loss
+
+
+def compute_dtm(img_gt, pred):
+    """
+    compute the distance transform map of foreground in mask
+    Args:
+        img_gt: Ground truth of the image, (b, h, w)
+        pred: Predictions of the segmentation head after softmax, (b, c, h, w)
```
Comment (xiexinch) [inline mmseg/models/losses/huasdorff_distance_loss.py:19]: Need to add type hints.

**Review by xiexinch — COMMENTED** [review]
We need to add some unit tests.

**Comment by vansin** [comment]
@jinxianwei  can refer to this document for writing unit tests (https://mmsig.openmmlab.com/general/unitest).

File: mmseg/models/losses/huasdorff_distance_loss.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    return total_loss / num_class
+
+
+@MODELS.register_module()
+class HuasdorffDisstanceLoss(nn.Module):
+    """HuasdorffDisstanceLoss
+    Args:
+        reduction (str, optional): The method used to reduce the loss into
+            a scalar. Defaults to 'mean'.
```
[the remark this answers is not in the document]
Comment (jinxianwei) [reply mmseg/models/losses/huasdorff_distance_loss.py]: ok

**Review by xiexinch — APPROVED** [review]
